types/shuffle-array/index.d.ts

```diff
@@ -27,14 +28,15 @@ interface ShuffleArray {
      * arr - The given array.
      * options - Optional configuration options.
      */
-    <T>(arr: T[], options?: ShuffleOption): T[];
+    <T>(arr: T[], options?: ShuffleOptions): T[];
     /**
-     * Pick one or more random elements from the given array.
+     * Pick one or more random elements from the given array. If options.picks is
+     * omitted or === 1, a single element will be returned; otherwise an array.
      *
      * arr - The given array.
      * options - Optional configuration options.
      */
-    pick<T>(arr: T[], options?: Object): T[];
+    pick<T>(arr: ReadonlyArray<T>, options?: PickOptions): T | T[];
```

Comment:
This could probably be written as a series of overloads since it’s often statically analyzable what `option.picks` is:

```suggestion
    pick<T>(arr: readonly T[], options: PickOptions & { picks: 1 }): T;
    pick<T>(arr: readonly T[], options: PickOptions): T | T[];
    pick<T>(arr: readonly T[]): T;
```

It would need some tests. [Example](https://www.typescriptlang.org/play/#code/JYOwLgpgTgZghgYwgAgArAQawPIAczAD2IAzsgN4BQyNyuGmJA-AFzIgCuAtgEbQDclAL6VKAEwgIANnCgoYHEAgLE6DADwAVAHwAKWVDZy4Y4lICeyTQG0AugBpkhfEVJt0WPCtLIAZBTUsEjYARmQhAEo2TUEJaVl5RWVXQMwtPQMjCBMzSxsHJxdiYLQGL1cSKKtkAB8rO1jJGTlkBSVvVPT9KENkY1MQC3rbKpjxJoTkADdZZEz2bj4oBtEEYrAnEAgQtk5eaGQAXlTuqEdyVJKwyME10g3iCAAmXcWD4-osU4jb9eRFTAgQgAdxAr32UFqCwhdiOJwM50ubAAsnAwAALAB0UDgIFMXF0EXCP0oQA)